src/syntax/parser.rs

```diff
@@ -512,6 +512,12 @@ where
         } else {
             ps.reset_peek();
         }
+    } else {
+        if let ast::Expression::AttributeExpression { ref id, .. } = selector {
+            if id.name.starts_with('-') {
+                return error!(ErrorKind::ForbiddenAttributeExpression);
```

Comment:
There should be a similar check in the branch which returns a new `SelectExpression` (or an early return before it) but in the other direction: only attributes of private messages can be selectors ([EBNF](https://github.com/projectfluent/fluent/blob/06d20408febd5c50da1c6b6e173568b403c07ba5/spec/fluent.ebnf#L74)).

Reply:
We should also disallow `VariantExpressions` as selectors, as per the [EBNF](https://github.com/projectfluent/fluent/blob/06d20408febd5c50da1c6b6e173568b403c07ba5/spec/fluent.ebnf#L64).